src/app/data-model-provider/MetadataTypes.h

```diff
@@ -87,15 +98,98 @@ enum class AttributeQualityFlags : uint32_t
 struct AttributeEntry
 {
     AttributeId attributeId;
-    BitFlags<AttributeQualityFlags> flags;
 
-    // read/write access will be missing if read/write is NOT allowed
-    //
-    // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    // Constructor
+
+    StartBitFieldInit // Disabling '-Wconversion' & '-Wconversion'
+
+        constexpr AttributeEntry(
+            AttributeId id = 0,                                                                 // attributeId initial value,
+                                                                                                // this could be altered later
+            AttributeQualityFlags attrQualityFlags     = static_cast<AttributeQualityFlags>(0), // mask.flags initial value
+            std::optional<Access::Privilege> readPriv  = static_cast<Access::Privilege>(0),     // mask.readPrivilege initial value
+            std::optional<Access::Privilege> writePriv = static_cast<Access::Privilege>(0)      // mask.writePrivilege initial value
+            ) :
+        attributeId{ id },
+        mask{ to_underlying(attrQualityFlags) & ((1 << 7) - 1), // Narrowing expression to 7 bits
+              to_underlying(readPriv.value_or(static_cast<Access::Privilege>(0))) &
+                  ((1 << 5) - 1), // Narrowing expression to 5 bits
+              to_underlying(writePriv.value_or(static_cast<Access::Privilege>(0))) & ((1 << 5) - 1) }
+    // Narrowing expression to 5 bits
+    {}
+
+    EndBitFieldInit // Enabling '-Wconversion' & '-Wconversion'
+
+        // Getter for mask.readPrivilege
+        constexpr std::optional<Access::Privilege>
+        GetReadPrivilege() const
+    {
+        if (ReadAllowed())
+        {
+            return static_cast<Access::Privilege>(mask.readPrivilege);
+        }
+        else
+        {
+            return std::nullopt;
+        }
```

Comment:
```suggestion
        return std::nullopt;
```

No `else` after return. Please update throughout.

Reply:
Fixed.